language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java

```diff
@@ -0,0 +1,138 @@
+        // a function call used in a local variable declaration, since it produces syntax errors.
+        return context.currentSyntaxTree().isPresent() && context.currentSemanticModel().isPresent()
+                && CodeActionNodeValidator.validate(context.nodeAtRange()) && 
+                !(node.kind() == SyntaxKind.MAPPING_CONSTRUCTOR && node.parent().kind() == SyntaxKind.TABLE_CONSTRUCTOR)
+                && !(node.kind() == SyntaxKind.FUNCTION_CALL && node.parent().kind() == SyntaxKind.LOCAL_VAR_DECL);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<CodeAction> getCodeActions(CodeActionContext context,
+                                           RangeBasedPositionDetails posDetails) {
+        
+        Node node = posDetails.matchedCodeActionNode(); 
+        String varName = getLocalVarName(context);
+        String value = node.toSourceCode().strip();
+        LineRange replaceRange = node.lineRange();
+        Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get().typeOf(node);
+        if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+            return Collections.emptyList();
+        }
+
+        Node statementNode = getStatementNode(node);
+        if (statementNode == null) {
+            return Collections.emptyList();
+        }
+        String paddingStr = StringUtils.repeat(" ", statementNode.lineRange().startLine().offset());
+        String varDeclStr = String.format("%s %s = %s;%n%s", typeSymbol.get().signature(), varName, value, paddingStr);
+        Position varDeclPos = new Position(statementNode.lineRange().startLine().line(), 
+                statementNode.lineRange().startLine().offset());
+        TextEdit varDeclEdit = new TextEdit(new Range(varDeclPos, varDeclPos), varDeclStr);
+        TextEdit replaceEdit = new TextEdit(new Range(PositionUtil.toPosition(replaceRange.startLine()),
+                PositionUtil.toPosition(replaceRange.endLine())),  varName);
+
+        return Collections.singletonList(CodeActionUtil.createCodeAction(CommandConstants.EXTRACT_TO_VARIABLE, 
+                List.of(varDeclEdit, replaceEdit), context.fileUri(), CodeActionKind.RefactorExtract));
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    private Node getStatementNode(Node node) {
+        Node statementNode = node;
+        while (statementNode != null && !(statementNode instanceof StatementNode)
+                && !(statementNode instanceof ModuleMemberDeclarationNode)) {
+            statementNode = statementNode.parent();
+        }
+        return statementNode;
+    }
+
+    private String getLocalVarName(CodeActionContext context) {
+        Position pos = context.range().getEnd();
+        Set<String> allNames = context.visibleSymbols(new Position(pos.getLine(), pos.getCharacter())).stream()
```

Comment:
You can directly use, context.range().getEnd() inside the visibleSymbols method rather than creating a Position